prime-router/src/main/kotlin/cli/tests/HistoryApiTest.kt

```diff
@@ -0,0 +1,233 @@
+    // CoolTest constructor.  This is a lotta work.
+    /**
+     * The access token left by a previous login command as specified by the command line parameters
+     */
+    fun getAccessToken(environment: Environment): String {
+        return if (environment.oktaApp == null) {
+            "dummy"
+        } else {
+            OktaCommand.fetchAccessToken(environment.oktaApp)
+                ?: abort(
+                    "Cannot run test $name.  Invalid access token. " +
+                        "Run ./prime login to fetch/refresh your access token for the $environment environment."
+                )
+        }
+    }
+
+    /**
+     * Create some fake history, so we have something to query for.
+     * @return null on failure.  Otherwise returns the list of ReportIds created.
+     */
+    private fun submitTestData(environment: Environment, options: CoolTestOptions): Set<ReportId>? {
+        val receivers = listOf<Receiver>(csvReceiver)
+        val counties = receivers.map { it.name }.joinToString(",")
+        val fakeItemCount = receivers.size * options.items
+        ugly("Starting $name test: Submitting ${options.submits} reports, each going to to $counties")
+        val file = FileUtilities.createFakeFile(
+            metadata,
+            settings,
+            historyTestSender,
+            fakeItemCount,
+            receivingStates,
+            counties,
+            options.dir,
+        )
+        echo("Created datafile $file")
+        // Now send it to ReportStream over and over
+        val reportIds = (1..options.submits).map {
+            val (responseCode, json) =
+                HttpUtilities.postReportFile(
+                    environment,
+                    file,
+                    historyTestSender,
+                    options.asyncProcessMode,
+                    options.key,
+                    payloadName = "$name ${status.description}",
+                )
+            echo("Response to POST: $responseCode")
+            if (responseCode != HttpURLConnection.HTTP_CREATED) {
+                bad("***$name Test FAILED***:  response code $responseCode")
+                return null // failure
+            }
+            val reportId = getReportIdFromResponse(json)
+                ?: let {
+                    bad("***$name Test FAILED***: A report ID came back as null")
+                    return null
+                }
+            echo("Id of submitted report: $reportId")
+            reportId
+        }.toSet()
+        return reportIds
+    }
+
+    /**
+     * Submit a query to the History API, get a response, and do very basic sanity checks on it.
+     * @return Pair(pass/fail status, json body string)
+     * The json body may be null even if the test passed, in cases of an expected error code.
+     */
+    private fun historyApiQuery(testCase: SubmissionAPITestCase): Pair<Boolean, String?> {
+        val (request, response, result) = Fuel.get(testCase.path, testCase.parameters)
```

Comment:
Please remove the "request" val (replace it with '_') since it is never used.

Reply:
done.